qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -574,83 +620,140 @@ def acquire(self, mode=None, samples_per_record=None,
+        sample_type = ctypes.c_uint8
+        if bytes_per_sample > 1:
+            sample_type = ctypes.c_uint16
+
         self.clear_buffers()
         # make sure that allocated_buffers <= buffers_per_acquisition
         if (self.allocated_buffers._get_byte() >
                 self.buffers_per_acquisition._get_byte()):
-            print("'allocated_buffers' should be smaller than or equal to"
-                  "'buffers_per_acquisition'. Defaulting 'allocated_buffers' to"
-                  "" + str(self.buffers_per_acquisition._get_byte()))
+            logger.warning("'allocated_buffers' should be <= "
+                            "'buffers_per_acquisition'. Defaulting 'allocated_buffers'"
+                            " to " + str(self.buffers_per_acquisition._get_byte()))
             self.allocated_buffers._set(
                 self.buffers_per_acquisition._get_byte())
 
         allocated_buffers = self.allocated_buffers._get_byte()
 
         for k in range(allocated_buffers):
             try:
-                self.buffer_list.append(Buffer(bps, samples_per_buffer,
-                                               number_of_channels))
+                self.buffer_list.append(Buffer(sample_type, bytes_per_buffer))
             except:
                 self.clear_buffers()
                 raise
 
         # post buffers to Alazar
-        for buf in self.buffer_list:
-            self._call_dll('AlazarPostAsyncBuffer',
-                           self._handle, buf.addr, buf.size_bytes)
-        self.allocated_buffers._set_updated()
-
-        # -----start capture here-----
-        acquisition_controller.pre_start_capture()
-        # call the startcapture method
-        self._call_dll('AlazarStartCapture', self._handle)
-
-        acquisition_controller.pre_acquire()
-        # buffer handling from acquisition
-        buffers_completed = 0
-        buffer_timeout = self.buffer_timeout._get_byte()
-        self.buffer_timeout._set_updated()
-
-        buffer_recycling = (self.buffers_per_acquisition._get_byte() >
-                            self.allocated_buffers._get_byte())
-
-        while buffers_completed < self.buffers_per_acquisition._get_byte():
-            buf = self.buffer_list[buffers_completed % allocated_buffers]
-
-            self._call_dll('AlazarWaitAsyncBufferComplete',
-                           self._handle, buf.addr, buffer_timeout)
-
-            # TODO(damazter) (C) last series of buffers must be handled
-            # exceptionally
-            # (and I want to test the difference) by changing buffer
-            # recycling for the last series of buffers
-
-            # if buffers must be recycled, extract data and repost them
-            # otherwise continue to next buffer
-
-            if buffer_recycling:
-                acquisition_controller.handle_buffer(buf.buffer)
+        try:
+            for buf in self.buffer_list:
                 self._call_dll('AlazarPostAsyncBuffer',
```

Comment:
When performing long acquisitions, AlazarPostAsyncBuffer has sometimes return an ApiFailed error. I haven't figured out why yet. Anyone else experienced this as well? If so, does this PR fix this issue? Otherwise, would it make sense to add a try except statement here that frees the buffer memory?